docs/make_work.jl

```diff
@@ -167,31 +167,18 @@ function sanitize(a::AbstractString, n::AbstractString)
   return joinpath(b[1], b[3:end]...)
 end
 
-function sanitize_dst(a::AbstractString, n::AbstractString)
-  b = splitpath(replace(a, Regex(".*$(n)") => n))
-  return joinpath("build", b[1], b[3:end]...)
-end
-
-bla = normpath(joinpath(dirname(pathof(Hecke)), "..", "docs", "src"))
-const hecke = sanitize(bla, "Hecke")
+sanitize_dst(a::AbstractString, n::AbstractString) = joinpath("build", sanitize(a, n))
 
-bla = normpath(joinpath(dirname(pathof(Nemo)), "..", "docs", "src"))
-const nemo = sanitize(bla, "Nemo")
-
-bla = normpath(joinpath(dirname(pathof(AbstractAlgebra)), "..", "docs", "src"))
-const aa = sanitize(bla, "AbstractAlgebra")
+const hecke = joinpath("Hecke.jl", "src")
+const nemo = joinpath("Nemo.jl", "src")
+const aa = joinpath("AbstractAlgebra.jl", "src")
```

Comment:
```suggestion
const aa = joinpath("AbstractAlgebra.jl", "docs", "src")
```

Reply:
No, then it won't work locally. I guess @fieker had a reason for the code as it was -- I just wish there were some comments explaining why it is like that (but I am guessing: trial and error 😂 ). I'll redo it.

Reply:
But was it not `$pkg/docs/src` before?

Reply:
Nope, not when run on my machine - the PR was tested and fully working on my machine before submitted... I'll dig to try and retrace Claus' steps

Reply:
I mean in the code, I thought this pointed to the docs. I am probably misunderstanding something.

Reply:
Same here: the way I read the code does not match with what I observe is shown by `println(nemo)` etc. -- clearly I am misunderstanding something, too. Weird.

Reply:
Here in CI it prints:
```
hecke = Hecke/docs/src
nemo = Nemo/docs/src
aa = AbstractAlgebra/docs/src
```
On my computer, where all three packages are dev'ed, it prints:
```
hecke = Hecke.jl/src
nemo = Nemo.jl/src
aa = AbstractAlgebra.jl/src
```
(note the added `.jl`).

Reply:
(and in both cases, the docs build and seem to work fine)

Reply:
When adding AA as usual, I get this:
```
julia> normpath(dirname(pathof(AbstractAlgebra)), "..", "docs", "src")
"/Users/mhorn/.julia/packages/AbstractAlgebra/Boo1X/docs/src"

julia> pkgdocprefix(AbstractAlgebra)
"AbstractAlgebra/docs/src"
```

When I dev a clone of AA, I get this:
```
julia> normpath(dirname(pathof(AbstractAlgebra)), "..", "docs", "src")
"/Users/mhorn/Projekte/OSCAR/AbstractAlgebra.jl/docs/src"

julia> pkgdocprefix(AbstractAlgebra)
"AbstractAlgebra.jl/src"
```
With `] dev AbstractAlgebra`, I get this:
```
julia> normpath(dirname(pathof(AbstractAlgebra)), "..", "docs", "src")
"/Users/mhorn/.julia/dev/AbstractAlgebra/docs/src"

julia> pkgdocprefix(AbstractAlgebra)
"AbstractAlgebra/src"
```

So all in all, the problem is that `sanitize` is not quite correct, and so we get weird results.

But all in all, I don't understand why we jump through all these hoops and patch Documenter, instead of manually copying the relevant .md files from AA / Nemo / Hecke into Oscar's `docs/src/` directory? I think Claus and Rafael looked into this, so there might be something I am missing; but I just hacked together a variant of that, and it seems much simpler than what we currently do, yet in a quick local tests *seems* to produce identical results. But maybe it doesn't, actually? I'll submit a separate PR for that.

Reply:
See PR #524

Reply:
whatever works, works. I was (am) of teh understanding that the package directories (docs/src) are readonly (or supposed to be readonly), thus the idea to pick where neccessary. Also to reduce the possibility of anyone editing the Hecke/Nemo/AA pages outside their repo.
But I am not married to "my" solution, it was also done just to show it can be done...